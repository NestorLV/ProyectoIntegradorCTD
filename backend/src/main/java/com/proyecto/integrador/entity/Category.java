package com.proyecto.integrador.entity;

import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.*;
import java.util.Date;
@Entity
@Getter
@Setter
@ToString
@Table(name = "category")
public class Category {
    @Id
    @Column(name = "id", nullable = false)
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    @Column(name = "title", nullable = false)
    private String title;
    @Column(name = "description", nullable = false)
    private String description;
    @Column(name = "url", nullable = false)
    private String url;

    public Category() {
    }

    public Category(Integer id, String title, String description, String url) {
        this.id = id;
        this.title = title;
        this.description = description;
        this.url = url;
    }

    public Category(String title, String description, String url) {
        this.title = title;
        this.description = description;
        this.url = url;
    }
}

