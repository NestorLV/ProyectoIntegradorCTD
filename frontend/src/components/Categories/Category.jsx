import React from 'react';
import Styles from './styles.module.css';

function Category(props) {
    const { imageUrl, title, description } = props;
    return (       
        <div className={Styles.categoryCard}>
            <img className= {Styles.categoryImage} src={imageUrl} alt="imagen 1" />
            <div className={Styles.categoryType}>
                <h3>{title}</h3>
                <p>{description}</p>
            </div>
        </div >        
    )                        
}

export default Category;