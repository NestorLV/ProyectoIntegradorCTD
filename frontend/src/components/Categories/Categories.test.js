import React from "react";
import { shallow } from "enzyme";
import ReactDOM from 'react-dom';
import "@testing-library/jest-dom";
import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { createSerializer } from 'enzyme-to-json';
import Category from "./Category";
import index from "./index";

Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

let wrapper;
let props = {handleCategory:jest.fn(), category: {id:1, name:"test", description:"test", image:"test", products:[]}, key:1,
title:"Hoteles", imageUrl:{url:"test"}, description:"test"};                                                                    



describe("<Category />", () => {
    beforeEach(() => {
        wrapper = shallow(<Category {...props}/>); 
    });

    it("Renderiza el componente correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    }); 

    it('Deberia disparar el evento handleCategory al hacer click', () => {
        wrapper = shallow(<Category {...props}/>);        
        
        wrapper.find('div.categoryCard').first().props().onClick()
        expect(props.handleCategory).toBeCalled();        
    });

    it('Debería renderizar la imagen con la url que viene por props', () => {       
        expect(wrapper.find('img').props().src).toBe(props.imageUrl);
    })

    it('Debería renderizar el componente h3 con el titulo title de props', () => {
        expect(wrapper.find('h3').text()).toBe(props.title);
    })

    it('Debería renderizar el componente p con la descripción description de props', () => {
        expect(wrapper.find('p').text()).toBe(props.description);
    })
});