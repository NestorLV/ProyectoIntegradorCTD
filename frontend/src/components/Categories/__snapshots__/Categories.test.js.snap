// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Category /> Renderiza el componente correctamente 1`] = `
<div
  className="categoryCard"
  onClick={[Function]}
>
  <img
    alt="imagen 1"
    className="categoryImage"
    src={
      Object {
        "url": "test",
      }
    }
  />
  <div
    className="categoryType"
  >
    <h3>
      Hoteles
    </h3>
    <p>
      test
    </p>
  </div>
</div>
`;
